test/client/ClientServiceTest.java

```diff
@@ -0,0 +1,95 @@
+package client;
+
+import static org.junit.Assert.assertEquals;
+
+import entities.Client;
+import enums.IocServicesEnum;
+import enums.StatusEnum;
+import exceptions.services.EntityNotFoundException;
+import exceptions.services.IdNullPointerException;
+import exceptions.services.NullEntityException;
+import ioc.Ioc;
+import java.util.List;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import services.implementation.ClientServiceImpl;
+import util.fixtures.ClientServiceFixtures;
+
+public class ClientServiceTest {
+  private ClientServiceImpl clientService;
+
+  @Before
+  public void setup() {
+    clientService = Ioc.getInstance().get(IocServicesEnum.CLIENT_SERVICE_INSTANCE);
+  }
+
+  @Test(expected = IdNullPointerException.class)
+  public void findById_ShouldThrowIdNullPointerException() {
+    clientService.findById(null);
+  }
+
+  @Test(expected = EntityNotFoundException.class)
+  public void findById_ShouldThrowEntityNotFoundException() {
+    clientService.findById(10L);
+  }
+
+  @Test
+  public void findById_ShouldReturnOptionalOfClient() {
+    Client client = ClientServiceFixtures.buildClient(null);
+    clientService.insert(client);
+    assertEquals(Optional.of(client), clientService.findById(client.getId()));
+  }
+
+  @Test(expected = EntityNotFoundException.class)
+  public void findById_ShouldThrowEntityNotFoundException_IfStatusIsDeleted() {
+    Client aux = new Client();
+    aux.setStatus(StatusEnum.DELETED);
+    Client client = ClientServiceFixtures.buildClient(aux);
+    clientService.insert(client);
+    assertEquals(Optional.of(client), clientService.findById(client.getId()));
```

Comment:
If this os supposed to return an exception remove this

Reply:
solved